### PR TITLE
introduce AT modifier

### PR DESCRIPTION
**Pull request by Nexucis**
Linked to the PRs:
* https://github.com/prometheus/prometheus/pull/8121
* https://github.com/prometheus/prometheus/pull/8425

Signed-off-by: Augustin Husson <husson.augustin@gmail.com>

### CODE REVIEW

File: test/expression.txt [hunk 1]
```diff
@@ -517,3 +517,130 @@ PromQL(
     )
   )
 )
+
+# Simple At start
+
+foo @ start()
```
Comment (juliusv) [inline test/expression.txt:523]: Could you add tests (and if necessary, code) for whitespace around the parens, like `foo @ start ( )`? Those should work in Prometheus itself, as the parentheses are separate tokens. Maybe they should be considered separate tokens here as well?

Comment (Nexucis) [reply test/expression.txt:523]: Maybe it will be easier to consider them as a separate token indeed.

**Comment by juliusv** [comment]
:+1: Thanks!
